### ISSUE TITLE
Opencomputers - Botania - item duplication

### ISSUE THREAD

**Issue by Bladowicz** [MightyPirates/OpenComputers#2370]
**TL;DR**

Improper robot's interaction with Botania block causes item duplication.

**Long story**

Im playing Minecraft version 1.10.2 with modpack SkyFactory 3 in version 3.0.8 which contains Opencomputers in version 1.6.2.7 and Botania in version 1.9.343.
I was trying to make automatic system for rune crafting. To do that i needed to place items on Rune Altar, then use wand .. etc. It looked like a good task for my first robot and my first lua script.
I made myself a proper robot, and decided to make a step by step proof of concept run with REPL. While first step, placing, worked correctly, the second step, removing items from altar, did not.
Removing items from altar requires shift+rmb with empty hand so i used
```lua
component.robot.use(sides.forward, _, true)
```

This resulted with:
- Item is still on the altar
- Item is in robots inventory
- Exception is raised

After few times i had multiple items.
Since Rune Altar allows any to be placed, any item can be duplicated with very little effort, and i think it may be major problem on some servers.

**Comment by Vexatos** [comment]
>Exception is raised

It might be useful to tell us what exactly it looks like.

**Comment by magik6k** [comment]
Can confirm:

```
[16:38:25] [Server thread/WARN] [OpenComputers]: Unexpected error in Lua callback.
java.lang.NullPointerException
	at net.minecraftforge.items.wrapper.PlayerMainInvWrapper.insertItem(PlayerMainInvWrapper.java:50) ~[PlayerMainInvWrapper.class:?]
	at net.minecraftforge.items.ItemHandlerHelper.insertItem(ItemHandlerHelper.java:40) ~[ItemHandlerHelper.class:?]
	at net.minecraftforge.items.ItemHandlerHelper.insertItemStacked(ItemHandlerHelper.java:102) ~[ItemHandlerHelper.class:?]
	at net.minecraftforge.items.ItemHandlerHelper.giveItemToPlayer(ItemHandlerHelper.java:169) ~[ItemHandlerHelper.class:?]
	at net.minecraftforge.items.ItemHandlerHelper.giveItemToPlayer(ItemHandlerHelper.java:144) ~[ItemHandlerHelper.class:?]
	at vazkii.botania.common.core.helper.InventoryHelper.withdrawFromInventory(InventoryHelper.java:77) ~[InventoryHelper.class:?]
	at vazkii.botania.common.block.mana.BlockRuneAltar.func_180639_a(BlockRuneAltar.java:76) ~[BlockRuneAltar.class:?]
	at li.cil.oc.server.agent.Player$$anonfun$activateBlockOrUseItem$1.apply(Player.scala:208) ~[Player$$anonfun$activateBlockOrUseItem$1.class:?]
	at li.cil.oc.server.agent.Player$$anonfun$activateBlockOrUseItem$1.apply(Player.scala:191) ~[Player$$anonfun$activateBlockOrUseItem$1.class:?]
	at li.cil.oc.server.agent.Player.callUsingItemInSlot(Player.scala:438) ~[Player.class:?]
	at li.cil.oc.server.agent.Player.activateBlockOrUseItem(Player.scala:191) ~[Player.class:?]
	at li.cil.oc.server.component.Agent$$anonfun$use$1.apply(Agent.scala:205) ~[Agent$$anonfun$use$1.class:?]
	at li.cil.oc.server.component.Agent$$anonfun$use$1.apply(Agent.scala:195) ~[Agent$$anonfun$use$1.class:?]
	at scala.collection.immutable.List.foreach(List.scala:383) ~[List.class:?]
	at li.cil.oc.server.component.Agent$class.use(Agent.scala:195) ~[Agent$class.class:?]
	at li.cil.oc.server.component.Robot.use(Robot.scala:29) ~[Robot.class:?]
	at generated.li.cil.oc.CallWrapper_li_cil_oc_server_component_Robot_use.call(Unknown Source) ~[?:?]
	at li.cil.oc.server.machine.Callbacks$ComponentCallback.apply(Callbacks.scala:123) ~[Callbacks$ComponentCallback.class:?]
	at li.cil.oc.server.network.Component$class.invoke(Component.scala:112) ~[Component$class.class:?]
	at li.cil.oc.server.network.Network$ComponentConnectorBuilder$$anon$1.invoke(Network.scala:631) ~[Network$ComponentConnectorBuilder$$anon$1.class:?]
	at li.cil.oc.server.machine.Machine.invoke(Machine.scala:358) ~[Machine.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.apply(ComponentAPI.scala:78) ~[ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.apply(ComponentAPI.scala:78) ~[ComponentAPI$$anonfun$initialize$5$$anonfun$apply$6.class:?]
	at li.cil.oc.server.machine.luac.NativeLuaArchitecture.invoke(NativeLuaArchitecture.scala:55) [NativeLuaArchitecture.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5.apply(ComponentAPI.scala:78) [ComponentAPI$$anonfun$initialize$5.class:?]
	at li.cil.oc.server.machine.luac.ComponentAPI$$anonfun$initialize$5.apply(ComponentAPI.scala:74) [ComponentAPI$$anonfun$initialize$5.class:?]
	at li.cil.oc.util.ExtendedLuaState$ExtendedLuaState$$anon$1.invoke(ExtendedLuaState.scala:24) [ExtendedLuaState$ExtendedLuaState$$anon$1.class:?]
	at li.cil.repack.com.naef.jnlua.LuaState.lua_pcall(Native Method) [LuaState.class:?]
	at li.cil.repack.com.naef.jnlua.LuaState.call(LuaState.java:681) [LuaState.class:?]
	at li.cil.oc.server.machine.luac.NativeLuaArchitecture.runSynchronized(NativeLuaArchitecture.scala:177) [NativeLuaArchitecture.class:?]
	at li.cil.oc.server.machine.Machine.update(Machine.scala:562) [Machine.class:?]
	at li.cil.oc.common.EventHandler$$anonfun$onServerTick$2.apply(EventHandler.scala:177) [EventHandler$$anonfun$onServerTick$2.class:?]
	at li.cil.oc.common.EventHandler$$anonfun$onServerTick$2.apply(EventHandler.scala:175) [EventHandler$$anonfun$onServerTick$2.class:?]
	at scala.collection.mutable.HashSet.foreach(HashSet.scala:78) [scala-library-2.11.1.jar:?]
	at li.cil.oc.common.EventHandler$.onServerTick(EventHandler.scala:175) [EventHandler$.class:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_870_EventHandler$_onServerTick_ServerTickEvent.invoke(.dynamic) [?:?]
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) [ASMEventHandler.class:?]
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185) [EventBus.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onPreServerTick(FMLCommonHandler.java:274) [FMLCommonHandler.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:602) [MinecraftServer.class:?]
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:149) [bzl.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_66]

```

**Comment by Vexatos** [comment]
Could someone please check whether it is fixed in [build 87](http://ci.cil.li/job/OpenComputers-dev-MC1.10/87/)?

**Comment by Bladowicz** [comment]
Updating SkyFactory's version of Opencomputers to build 87 solved this problem for me. :+1:
